pkg/cmd/server/origin/controller/serviceaccount.go

```diff
@@ -0,0 +1,129 @@
+package controller
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/golang/glog"
+
+	"k8s.io/client-go/util/cert"
+	kapiv1 "k8s.io/kubernetes/pkg/api/v1"
+	sacontroller "k8s.io/kubernetes/pkg/controller/serviceaccount"
+	"k8s.io/kubernetes/pkg/serviceaccount"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	"github.com/openshift/origin/pkg/cmd/server/crypto"
+	serviceaccountcontrollers "github.com/openshift/origin/pkg/serviceaccounts/controllers"
+)
+
+type ServiceAccountControllerOptions struct {
+	ManagedNames []string
+}
+
+func (c *ServiceAccountControllerOptions) RunController(ctx ControllerContext) (bool, error) {
+	if len(c.ManagedNames) == 0 {
+		return true, fmt.Errorf("skipped starting Service Account Manager, no managed names specified")
+	}
+
+	options := sacontroller.DefaultServiceAccountsControllerOptions()
+	options.ServiceAccounts = []kapiv1.ServiceAccount{}
+
+	for _, saName := range c.ManagedNames {
+		sa := kapiv1.ServiceAccount{}
+		sa.Name = saName
+
+		options.ServiceAccounts = append(options.ServiceAccounts, sa)
+	}
+
+	//REBASE: add new args to NewServiceAccountsController
+	go sacontroller.NewServiceAccountsController(
+		ctx.DeprecatedOpenshiftInformers.KubernetesInformers().Core().V1().ServiceAccounts(),
+		ctx.DeprecatedOpenshiftInformers.KubernetesInformers().Core().V1().Namespaces(),
+		ctx.ClientBuilder.ClientOrDie(bootstrappolicy.InfraServiceAccountControllerServiceAccountName),
+		options).Run(1, ctx.Stop)
+
+	return true, nil
+}
+
+type ServiceAccountTokensControllerOptions struct {
+	ServiceAccountConfig     configapi.ServiceAccountConfig
+	ServiceServingCertConfig configapi.ServiceServingCert
+}
+
+func (c *ServiceAccountTokensControllerOptions) RunController(ctx ControllerContext) (bool, error) {
+	if len(c.ServiceAccountConfig.PrivateKeyFile) == 0 {
+		return true, fmt.Errorf("skipped starting Service Account Token Manager, no private key specified")
```

Comment:
`return false, nil`.  The bool indicates that this controller was enabled, the error means we couldn't start the controller when we were supposed to.  So returning false without an error means "skipped" and return an error means, "couldn't fullfill intent, crashloop."